src/editor/Model/Tokenizer.re

```diff
@@ -8,118 +8,93 @@ open Oni_Core;
 open Oni_Core.Types;
 open Oni_Extensions;
 
+open CamomileLibrary;
+
 type t = {
   text: string,
   startPosition: Index.t,
   endPosition: Index.t,
   color: Color.t,
 };
 
+let space = UChar.of_char(' ');
+let tab = UChar.of_char('\t');
+let cr = UChar.of_char('\r');
+let lf = UChar.of_char('\n');
+
 let _isWhitespace = c => {
-  c == ' ' || c == '\t' || c == '\r' || c == '\n';
+  UChar.eq(space, c)
+  || UChar.eq(tab, c)
+  || UChar.eq(cr, c)
+  || UChar.eq(lf, c);
 };
 
 let _isNonWhitespace = c => !_isWhitespace(c);
 
-let _moveToNextMatchingToken = (f, str, startIdx) => {
-  let idx = ref(startIdx);
-  let length = Zed_utf8.length(str);
-  let found = ref(false);
-
-  while (idx^ < length && ! found^) {
-    let c = str.[idx^];
-    found := f(c);
-
-    if (! found^) {
-      idx := idx^ + 1;
-    };
-  };
-
-  idx^;
-};
-
-let _moveToNextWhitespace = _moveToNextMatchingToken(_isWhitespace);
-let _moveToNextNonWhitespace = _moveToNextMatchingToken(_isNonWhitespace);
+let tokenize:
```

Comment:
Add benchmark for this; compare with `master` implementation. This is on the rendering critical path.